src/utilities/alfalfa/AlfalfaPoint.cpp

```diff
@@ -4,15 +4,21 @@
 #include <boost/regex.hpp>
 #include <fmt/format.h>
 
+#include <memory>
+#include <string_view>
+
 namespace openstudio {
 namespace alfalfa {
+
+  static constexpr std::string_view ID_VALID_CHARS_MSG = "IDs can only contain letters, numbers, and the following special characters _-[]():";
```

Comment:
this is a global static constexpr variable. I could omit the `static`, and it'd still be `static` because it's declared at global scope and it's constexpr, but this is more explicit.

And so it initializes it in static storage (the `.rodata` section of the binary). The string_view is materialized in the binary.

see this short example: https://gcc.godbolt.org/z/o7veTMETP

and if you want to learn more about constexpr, I suggest C++ weekly, the start of this episode maybe: https://www.youtube.com/watch?v=IDQ0ng8RIqs